sdk/core/Azure.Core/tests/RecordSessionTests.cs

```diff
@@ -84,6 +84,63 @@ public void CanRoundtripSessionRecord(string body, string contentType)
             CollectionAssert.AreEqual(bodyBytes, deserializedRecord.Response.Body);
         }
 
+        [Test]
+        public void CanRoundTripEncodedJson()
```

Comment:
Would it make sense to either add the same test cases as on `CanRoundtripSessionRecord` or just modify that test to create a `MockRequest` to hold the content as you'd done here?

Reply:
Yeah, let's reuse CanRoundtripSessionRecord